sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -125,14 +127,16 @@ class StateOperatorProgress private[sql](
  * @since 2.1.0
  */
 @Evolving
-class StreamingQueryProgress private[sql](
+class StreamingQueryProgress private[spark](
   val id: UUID,
   val runId: UUID,
   val name: String,
   val timestamp: String,
   val batchId: Long,
   val batchDuration: Long,
   val durationMs: ju.Map[String, JLong],
+  // If `eventTime` is no longer read-only, need change
+  // `StreamingQueryProgressSerializer#deserialize` to return mutable Java Map
   val eventTime: ju.Map[String, String],
```

Comment:
There are 6 `ju.Map`,  `durationMs` and `observedMetrics` already use mutable Java Map, the remaining 4 Java Map should be read-only at present, but for safety, we can also directly return mutable Java Map in the deserialize method